src/parcsr_ls/ams.c

```diff
@@ -3444,8 +3444,8 @@ HYPRE_Int hypre_AMSSetup(void *solver,
                {
                   GGt = hypre_ParCSRMatMat(ams_data -> G, Gt);
                }
-#endif
                else
+#endif
```

Comment:
This is a good catch. I was wondering how could it work before (with correct result)? Have you seen results changed with this fix?

Reply:
Yes, it was a good catch by the compiler... Imagine trying to debug this... I haven't seen CPU results changing after this fix. Maybe because https://github.com/hypre-space/hypre/blob/292f62cb7afb1f3b56472527f84116bf8efffcde/src/parcsr_ls/ams.c#L3429 is not true in the scenarios we test? So the following else block is always executed...  I'm not 100% sure, but that seems the case.